public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx

```diff
@@ -146,6 +157,39 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
     return CoreApp.Dashboard;
   }
 
+  changeToSnapshotData = (frames: DataFrame[]) => {
+    const snapshot: DataFrameJSON[] = frames.map((f) => dataFrameToJSON(f));
+
+    const query: GrafanaQuery = {
+      refId: 'A',
+      queryType: GrafanaQueryType.Snapshot,
+      snapshot,
+      datasource: { uid: GRAFANA_DATASOURCE_NAME },
+    };
+
+    this.props.panel.updateQueries({
+      dataSource: { uid: GRAFANA_DATASOURCE_NAME },
+      queries: [query],
+    });
+
+    this.props.panel.refresh();
+
+    this.closeConfirmationModal();
+  };
+
```

Comment:
Can we move this to a function/file in the grafana datasource? 

And I think that function could handle showing the confirm modal. You can show a confirm modal from anywhere with simply this

```ts
 appEvents.publish(
        new ShowConfirmModalEvent({
          title: 'Delete',
          text: 'Are you sure you want to permanently delete your starred query?',
          yesText: 'Delete',
          icon: 'trash-alt',
          onConfirm: () => performDelete(query.id),
        })
      );
```

Don't want to add to much stuff to PanelStateWrapper that is too large already :)

Reply:
@mdvictor what does happen in the data grid panel if you cancel the update? Is the edit change reverted in the panel (without knowing it was cancelled).

Reply:
The confirmation modal makes more sense. I pushed a fix. As for the datagrid, the data won't change in the view with anything but the latest frame. If the update fails, then the frame would not change and there would be no updated cell on cell edit. I do need to make a few changes in the datagrid though to fully support this.

Even though the datagrid itself doesn't need to know whether the event was successful or not, I think there might be cases where the panel needs to know what the user selected on the modal, so I changed the `onUpdateData` and added back the boolean promise.